gtsfm/evaluation/metrics_report.py

```diff
@@ -125,22 +158,99 @@ def get_figures_for_metrics(metrics_group: GtsfmMetricsGroup) -> Tuple[str, str]
     """
     scalar_metrics = {}
     metrics_dict = metrics_group.get_metrics_as_dict()[metrics_group.name]
-
     # Separate the scalar metrics.
     for metric_name, value in metrics_dict.items():
         if isinstance(value, dict):
             # Metrics with a dict representation must contain a summary.
             if metrics.SUMMARY_KEY not in value:
                 raise ValueError(f"Metric {metric_name} does not contain a summary.")
-            # Add a scalar metric for mean of 1D distributions.
-            scalar_metrics["mean_" + metric_name] = value[metrics.SUMMARY_KEY]["mean"]
+            # Add a scalar metric for median of 1D distributions.
+            scalar_metrics["median_" + metric_name] = value[metrics.SUMMARY_KEY]["median"]
         else:
             scalar_metrics[metric_name] = value
     table = create_table_for_scalar_metrics(scalar_metrics)
     plots_fig = create_plots_for_distributions(metrics_dict)
     return table, plots_fig
 
 
+def add_summary(metric_name, metric_value, scalar_metrics):
```

Comment:
regarding both add_metric and add_summary. I dont think we mutate arguments anywhere in gtsfm, and try to be functional despite being in python land. @johnwlambert please correct me if that is not the case.

Reply:
so I think we dont need this as a separate function, but if you have other ways to make it functional, that also works.